From 06cc137447368dad77508bc7942b095621238741 Mon Sep 17 00:00:00 2001
From: Nightfox <cameronscottarmstrong@gmail.com>
Date: Sat, 25 May 2019 17:56:33 +0800
Subject: [PATCH 04/18] Ctrl - K added as key combination to comment lines out

---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index ff395e1..099cade 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -460,7 +460,7 @@ int main(int argc, const char *argv[])
           mDebugOutput.SetText( szError );
         }
       }
-      else if (Renderer::keyEventBuffer[i].ctrl && Renderer::keyEventBuffer[i].scanCode == '/') // Ctrl Slash (/)
+      else if (Renderer::keyEventBuffer[i].ctrl && (Renderer::keyEventBuffer[i].scanCode == '/' || Renderer::keyEventBuffer[i].scanCode == 'k')) // Ctrl/Cmd Slash (/) or Ctrl/Cmd-k
       {
         mShaderEditor.CommentSelection();
       }
-- 
2.30.0

