From 9c2a4bae2fb59d9278e05ae51f92cc5cd22124dd Mon Sep 17 00:00:00 2001
From: Nightfox <cameronscottarmstrong@gmail.com>
Date: Sat, 25 May 2019 17:56:33 +0800
Subject: [PATCH 05/21] Ctrl - K added as key combination to comment lines out

---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/main.cpp src/main.cpp
index eaf4e43..29f3969 100644
--- src/main.cpp
+++ src/main.cpp
@@ -444,7 +444,7 @@ int main(int argc, const char *argv[])
           mDebugOutput.SetText( szError );
         }
       }
-      else if (Renderer::keyEventBuffer[i].ctrl && Renderer::keyEventBuffer[i].scanCode == '/') // Ctrl Slash (/)
+      else if (Renderer::keyEventBuffer[i].ctrl && (Renderer::keyEventBuffer[i].scanCode == '/' || Renderer::keyEventBuffer[i].scanCode == 'k')) // Ctrl/Cmd Slash (/) or Ctrl/Cmd-k
       {
         mShaderEditor.CommentSelection();
       }
-- 
2.25.0

