From 6f596fce032527ae43432d2ccc5c1d7edf9824e8 Mon Sep 17 00:00:00 2001
From: Florent CURE <florent.cure@protonmail.com>
Date: Tue, 28 May 2019 09:56:37 +0100
Subject: [PATCH 12/18] Fix missing header for getenv()

---
 src/platform_linux/Config.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/platform_linux/Config.cpp b/src/platform_linux/Config.cpp
index 945c76b..24be680 100644
--- a/src/platform_linux/Config.cpp
+++ b/src/platform_linux/Config.cpp
@@ -5,6 +5,7 @@
 #include <vector>
 
 #include <unistd.h>
+#include <stdlib.h>
 
 
 namespace Config
-- 
2.30.0

