From 7a2661969e232b8d7c9ccdf2814831da87d3a39c Mon Sep 17 00:00:00 2001
From: Florent CURE <florent.cure@protonmail.com>
Date: Tue, 28 May 2019 09:39:43 +0100
Subject: [PATCH 13/21] Add missing headers

---
 src/platform_common/Config.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/platform_common/Config.cpp src/platform_common/Config.cpp
index 45913b1..0caa5b6 100644
--- src/platform_common/Config.cpp
+++ src/platform_common/Config.cpp
@@ -1,5 +1,8 @@
 #include "Config.h"
 
+#include <string>
+#include <fstream>
+
 #ifdef _WIN32
 #include <direct.h>
 #define getcwd _getcwd
-- 
2.25.0

