From 138e5a61629addc8bf3ecbad0331bb2ef067ca6c Mon Sep 17 00:00:00 2001
From: Florent CURE <florent.cure@protonmail.com>
Date: Tue, 28 May 2019 09:51:37 +0100
Subject: [PATCH 15/21] Fix read_file which didnt compile on some platform

---
 src/platform_common/Utils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/platform_common/Utils.cpp src/platform_common/Utils.cpp
index 8b67d13..101ce61 100644
--- src/platform_common/Utils.cpp
+++ src/platform_common/Utils.cpp
@@ -7,7 +7,7 @@ namespace Utils
 
 	bool read_file( const std::string& filename, std::vector< char >& out )
 	{
-		std::ifstream fp( filename );
+		std::ifstream fp( filename.c_str() );
 		if( !fp.is_open() ) return false;
 		fp.seekg( 0, std::ios::end );
 		size_t file_size_in_byte = fp.tellg();
-- 
2.25.0

