From fbc0c7b26b57b214d78262f171c5ad324a9d9156 Mon Sep 17 00:00:00 2001
From: Emmanuel Gil Peyrot <linkmauve@linkmauve.fr>
Date: Sat, 6 Oct 2018 12:43:46 +0200
Subject: [PATCH 16/18] Add an option to build against system-wide GLFW and
 GLEW

---
 CMakeLists.txt | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 493f674..3599ff0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,10 +32,11 @@ if (WIN32)
   option(BONZOMATIC_NDI "Enable NDI?" OFF)
 endif ()
 
-if (WIN32)
-  set(BONZOMATIC_WINDOWS_FLAVOR "GLFW" CACHE STRING "Windows renderer flavor selected at CMake configure time (DX11, DX9 or GLFW)")
-  set_property(CACHE BONZOMATIC_WINDOWS_FLAVOR PROPERTY STRINGS DX11 DX9 GLFW)
-endif ()
+option(BONZOMATIC_USE_SYSTEMWIDE_GLFW "Use the system-wide version of GLFW instead of the vendored one (this build option is not supported by upstream)?" OFF)
+option(BONZOMATIC_USE_SYSTEMWIDE_GLEW "Use the system-wide version of GLEW instead of the vendored one (this build option is not supported by upstream)?" OFF)
+
+set(BONZOMATIC_WINDOWS_FLAVOR "DX11" CACHE STRING "Windows renderer flavor selected at CMake configure time (DX11, DX9 or GLFW)")
+set_property(CACHE BONZOMATIC_WINDOWS_FLAVOR PROPERTY STRINGS DX11 DX9 GLFW) 
 
 if (NOT (UNIX AND (NOT APPLE))) #if not linux
   set(CMAKE_INSTALL_PREFIX ${CMAKE_BINARY_DIR})
@@ -74,8 +75,10 @@ if (APPLE OR UNIX OR (WIN32 AND (${BONZOMATIC_WINDOWS_FLAVOR} MATCHES "GLFW")))
   # GLFW
   # GLFW settings and project inclusion
   find_package(PkgConfig)
-  pkg_check_modules(GLFW glfw3)
-  if (GLFW_FOUND)
+  if (PKG_CONFIG_FOUND AND BONZOMATIC_USE_SYSTEMWIDE_GLFW)
+    pkg_check_modules(GLFW glfw3)
+  endif()
+  if (BONZOMATIC_USE_SYSTEMWIDE_GLFW AND GLFW_FOUND)
     set(BZC_PROJECT_INCLUDES ${BZC_PROJECT_INCLUDES} ${GLFW_INCLUDES})
     set(BZC_PROJECT_LIBS ${BZC_PROJECT_LIBS} ${GLFW_LIBRARIES})
   else()
@@ -100,8 +103,10 @@ if (APPLE OR UNIX OR (WIN32 AND (${BONZOMATIC_WINDOWS_FLAVOR} MATCHES "GLFW")))
 
   ##############################################################################
   # GLEW
-  pkg_check_modules(GLEW glew)
-  if (GLEW_FOUND)
+  if (PKG_CONFIG_FOUND AND BONZOMATIC_USE_SYSTEMWIDE_GLEW)
+    pkg_check_modules(GLEW glew)
+  endif()
+  if (BONZOMATIC_USE_SYSTEMWIDE_GLEW AND GLEW_FOUND)
     set(BZC_PROJECT_INCLUDES ${BZC_PROJECT_INCLUDES} ${GLEW_INCLUDES})
     set(BZC_PROJECT_LIBS ${BZC_PROJECT_LIBS} ${GLEW_LIBRARIES})
   else ()
-- 
2.30.0

